src/pages/journal/components/test-outcome/__tests__/test-outcome.spec.ts

```diff
@@ -167,5 +163,77 @@ describe('Test Outcome', () => {
       });
     });
 
+    describe('show rekey button', () => {
+      it('should show rekey button for a booked test if date is in past', () => {
+        component.slotDetail = testSlotDetail;
+
+        const dateTime = new Date();
+        dateTime.setDate(dateTime.getDate() - 1);
+
+        component.slotDetail.start = dateTime.toUTCString();
+        component.testStatus = TestStatus.Booked;
+        fixture.detectChanges();
+
+        const rekeyButton = fixture.debugElement.query(By.css('.mes-rekey-button'));
+
+        expect(rekeyButton).not.toBeNull();
+      });
+      it('should show rekey button for a resumed test if date is in past', () => {
+        component.slotDetail = testSlotDetail;
+
+        const dateTime = new Date();
+        dateTime.setDate(dateTime.getDate() - 1);
+
+        component.slotDetail.start = dateTime.toUTCString();
+        component.testStatus = TestStatus.Started;
+        fixture.detectChanges();
+
+        const rekeyButton = fixture.debugElement.query(By.css('.mes-rekey-button'));
+
+        expect(rekeyButton).not.toBeNull();
+      });
+      it('should hide rekey button for a booked test if date is today', () => {
+        component.slotDetail = testSlotDetail;
+
+        const dateTime = new Date();
+
+        component.slotDetail.start = dateTime.toUTCString();
+        component.testStatus = TestStatus.Booked;
+        fixture.detectChanges();
+
+        const rekeyButton = fixture.debugElement.query(By.css('.mes-rekey-button'));
+
+        expect(rekeyButton).toBeNull();
+      });
+      it('should hide rekey button for a resumed test if date is today', () => {
+        component.slotDetail = testSlotDetail;
+
+        const dateTime = new Date();
+
+        component.slotDetail.start = dateTime.toUTCString();
+        component.testStatus = TestStatus.Started;
+        fixture.detectChanges();
+
+        const rekeyButton = fixture.debugElement.query(By.css('.mes-rekey-button'));
+
+        expect(rekeyButton).toBeNull();
+      });
+      it('should display the rekey modal for a test today that has ended', () => {
+        component.slotDetail = testSlotDetail;
+
+        const dateTime = new Date();
+        dateTime.setHours(dateTime.getHours() - 2);
```

Comment:
Similar comment  var.setDate(getDate() -1) -> it might go to negative values depending on the date/time when the test is executed